From 0 Mon Sep 17 00:00:00 2001
From: "Fr. Br. George" <george@altlinux.org>
Date: Mon, 15 Oct 2012 15:44:28 +0400
Subject: [PATCH] bind-9.9.7-owl-warnings


diff --git a/bind/lib/isc/Makefile.in b/bind/lib/isc/Makefile.in
index defaced..defaced 100644
--- a/bind/lib/isc/Makefile.in
+++ b/bind/lib/isc/Makefile.in
@@ -61,7 +61,7 @@ OBJS =		@ISC_EXTRA_OBJS@ \
 		ratelimiter.@O@ refcount.@O@ region.@O@ regex.@O@ result.@O@ \
 		rwlock.@O@ \
 		safe.@O@ serial.@O@ sha1.@O@ sha2.@O@ sockaddr.@O@ stats.@O@ \
-		string.@O@ strtoul.@O@ symtab.@O@ task.@O@ taskpool.@O@ \
+		string.@O@ symtab.@O@ task.@O@ taskpool.@O@ \
 		tm.@O@ timer.@O@ version.@O@ \
 		${UNIXOBJS} ${NLSOBJS} ${THREADOBJS}
 SYMTBLOBJS =	backtrace-emptytbl.@O@
@@ -78,7 +78,7 @@ SRCS =		@ISC_EXTRA_SRCS@ \
 		parseint.c portset.c quota.c radix.c random.c \
 		ratelimiter.c refcount.c region.c regex.c result.c rwlock.c \
 		safe.c serial.c sha1.c sha2.c sockaddr.c stats.c string.c \
-		strtoul.c symtab.c task.c taskpool.c timer.c \
+		symtab.c task.c taskpool.c timer.c \
 		tm.c version.c
 
 LIBS =		@LIBS@
diff --git a/bind/lib/isc/unix/net.c b/bind/lib/isc/unix/net.c
index defaced..defaced 100644
--- a/bind/lib/isc/unix/net.c
+++ b/bind/lib/isc/unix/net.c
@@ -103,10 +103,10 @@ const struct in6_addr isc_net_in6addrloop = IN6ADDR_LOOPBACK_INIT;
 
 # if defined(WANT_IPV6)
 static isc_once_t 	once_ipv6only = ISC_ONCE_INIT;
-# endif
 
-# if defined(ISC_PLATFORM_HAVEIN6PKTINFO)
+#  if defined(ISC_PLATFORM_HAVEIN6PKTINFO)
 static isc_once_t 	once_ipv6pktinfo = ISC_ONCE_INIT;
+#  endif
 # endif
 #endif /* ISC_PLATFORM_HAVEIPV6 */
 
diff --git a/bind/lib/lwres/Makefile.in b/bind/lib/lwres/Makefile.in
index defaced..defaced 100644
--- a/bind/lib/lwres/Makefile.in
+++ b/bind/lib/lwres/Makefile.in
@@ -36,14 +36,14 @@ OBJS =		compat.@O@ context.@O@ \
 		getipnode.@O@ getnameinfo.@O@ getrrset.@O@ herror.@O@ \
 		lwbuffer.@O@ lwconfig.@O@ lwpacket.@O@ lwresutil.@O@ \
 		lwres_gabn.@O@ lwres_gnba.@O@ lwres_grbn.@O@ lwres_noop.@O@ \
-		lwinetaton.@O@ lwinetpton.@O@ lwinetntop.@O@ print.@O@
+		lwinetaton.@O@ lwinetpton.@O@ lwinetntop.@O@
 
 # Alphabetically
 SRCS =		compat.c context.c gai_strerror.c getaddrinfo.c gethost.c \
 		getipnode.c getnameinfo.c getrrset.c herror.c \
 		lwbuffer.c lwconfig.c lwpacket.c lwresutil.c \
 		lwres_gabn.c lwres_gnba.c lwres_grbn.c lwres_noop.c \
-		lwinetaton.c lwinetpton.c lwinetntop.c print.c
+		lwinetaton.c lwinetpton.c lwinetntop.c
 
 LIBS =		@LIBS@
 
