From 6f03c868a7dd873fe25de062e794e83b0e22768d Mon Sep 17 00:00:00 2001
From: - <->
Date: Mon, 15 Oct 2012 15:44:28 +0400
Subject: [PATCH] owl-warnings


diff --git a/bind/lib/isc/Makefile.in b/bind/lib/isc/Makefile.in
index bd8b109..877246d 100644
--- a/bind/lib/isc/Makefile.in
+++ b/bind/lib/isc/Makefile.in
@@ -58,7 +58,7 @@ OBJS =		@ISC_EXTRA_OBJS@ @ISC_PK11_O@ @ISC_PK11_RESULT_O@ \
 		ratelimiter.@O@ refcount.@O@ region.@O@ regex.@O@ result.@O@ \
 		rwlock.@O@ \
 		safe.@O@ serial.@O@ siphash.@O@ sha1.@O@ sha2.@O@ sockaddr.@O@ stats.@O@ \
-		string.@O@ strtoul.@O@ symtab.@O@ task.@O@ taskpool.@O@ \
+		string.@O@ symtab.@O@ task.@O@ taskpool.@O@ \
 		tm.@O@ timer.@O@ utf8.@O@ version.@O@ \
 		${UNIXOBJS} ${NLSOBJS} ${THREADOBJS}
 SYMTBLOBJS =	backtrace-emptytbl.@O@
@@ -77,7 +77,7 @@ SRCS =		@ISC_EXTRA_SRCS@ @ISC_PK11_C@ @ISC_PK11_RESULT_C@ \
 		parseint.c portset.c quota.c radix.c random.c ${CHACHASRCS} \
 		ratelimiter.c refcount.c region.c regex.c result.c rwlock.c \
 		safe.c serial.c siphash.c sha1.c sha2.c sockaddr.c stats.c string.c \
-		strtoul.c symtab.c task.c taskpool.c timer.c \
+		symtab.c task.c taskpool.c timer.c \
 		tm.c utf8.c version.c
 
 LIBS =		@ISC_OPENSSL_LIBS@ @LIBS@
diff --git a/bind/lib/lwres/Makefile.in b/bind/lib/lwres/Makefile.in
index fe39b36..985f771 100644
--- a/bind/lib/lwres/Makefile.in
+++ b/bind/lib/lwres/Makefile.in
@@ -27,14 +27,14 @@ OBJS =		compat.@O@ context.@O@ \
 		getipnode.@O@ getnameinfo.@O@ getrrset.@O@ herror.@O@ \
 		lwbuffer.@O@ lwconfig.@O@ lwpacket.@O@ lwresutil.@O@ \
 		lwres_gabn.@O@ lwres_gnba.@O@ lwres_grbn.@O@ lwres_noop.@O@ \
-		lwinetaton.@O@ lwinetpton.@O@ lwinetntop.@O@ print.@O@
+		lwinetaton.@O@ lwinetpton.@O@ lwinetntop.@O@
 
 # Alphabetically
 SRCS =		compat.c context.c gai_strerror.c getaddrinfo.c gethost.c \
 		getipnode.c getnameinfo.c getrrset.c herror.c \
 		lwbuffer.c lwconfig.c lwpacket.c lwresutil.c \
 		lwres_gabn.c lwres_gnba.c lwres_grbn.c lwres_noop.c \
-		lwinetaton.c lwinetpton.c lwinetntop.c print.c
+		lwinetaton.c lwinetpton.c lwinetntop.c
 
 LIBS =		@LIBS@
 
-- 
2.10.5

