From 0 Mon Sep 17 00:00:00 2001
From: "Fr. Br. George" <george@altlinux.org>
Date: Mon, 15 Oct 2012 15:44:28 +0400
Subject: [PATCH] bind-9.9.1-alt-ads-remove


diff --git a/bind/bin/named/main.c b/bind/bin/named/main.c
index defaced..defaced 100644
--- a/bind/bin/named/main.c
+++ b/bind/bin/named/main.c
@@ -905,6 +905,7 @@ setup(void) {
 		      ISC_LOG_NOTICE, "starting %s %s%s", ns_g_product,
 		      ns_g_version, saved_command_line);
 
+	/*
 	isc_log_write(ns_g_lctx, NS_LOGCATEGORY_GENERAL, NS_LOGMODULE_MAIN,
 		      ISC_LOG_NOTICE, "built with %s", ns_g_configargs);
 
@@ -927,6 +928,7 @@ setup(void) {
 		      ISC_LOG_NOTICE,
 		      "----------------------------------------------------");
 
+	*/
 	dump_symboltable();
 
 	/*
