--- bind-8.2.2_P5/src/bin/host/host.c.mx	Thu Feb  3 16:00:23 2000
+++ bind-8.2.2_P5/src/bin/host/host.c	Thu Feb  3 17:18:14 2000
@@ -225,6 +225,8 @@
 	int ncnames, ch;
 	int nkeychains, i;
 
+	gettype = 0;
+
 	dst_init();
 
 	if ((progname = strrchr(argv[0], '/')) == NULL)
@@ -269,11 +271,6 @@
 			usage("unrecogized switch");
 			/*NOTREACHED*/
 		}
-	}
-	if (gettype == 0) {
-		if (verbose)
-			printf ("Forcing `-t a' for signature trace.\n");
-		gettype = ns_t_a;
 	}
 	argc -= optind;
 	argv += optind;
