diff -ur bind-9.2.2.orig/bin/check/named-checkconf.8 bind-9.2.2/bin/check/named-checkconf.8
--- bind-9.2.2.orig/bin/check/named-checkconf.8	Fri Jun 29 04:04:54 2001
+++ bind-9.2.2/bin/check/named-checkconf.8	Thu Jun  5 11:19:17 2003
@@ -30,6 +30,7 @@
 chroot to \fIdirectory\fR so that include
 directives in the configuration file are processed as if
 run by a similarly chrooted named.
+By default, \fBnamed-checkconf\fR \fBchroot()\fR's to \fI@ROOT@\fR.
 .TP
 \fB-v\fR
 Print the version of the \fBnamed-checkconf\fR
diff -ur bind-9.2.2.orig/bin/check/named-checkconf.c bind-9.2.2/bin/check/named-checkconf.c
--- bind-9.2.2.orig/bin/check/named-checkconf.c	Fri Jul 27 20:45:26 2001
+++ bind-9.2.2/bin/check/named-checkconf.c	Wed Jun  4 15:49:04 2003
@@ -78,22 +78,12 @@
 	isc_mem_t *mctx = NULL;
 	isc_result_t result;
 	int exit_status = 0;
+	const char *chroot_dir = "@ROOT@";
 
 	while ((c = isc_commandline_parse(argc, argv, "t:v")) != EOF) {
 		switch (c) {
 		case 't':
-			result = isc_dir_chroot(isc_commandline_argument);
-			if (result != ISC_R_SUCCESS) {
-				fprintf(stderr, "isc_dir_chroot: %s\n",
-					isc_result_totext(result));
-				exit(1);
-			}
-			result = isc_dir_chdir("/");
-			if (result != ISC_R_SUCCESS) {
-				fprintf(stderr, "isc_dir_chdir: %s\n",
-					isc_result_totext(result));
-				exit(1);
-			}
+			chroot_dir = isc_commandline_argument;
 			break;
 
 		case 'v':
@@ -104,6 +94,18 @@
 			usage();
 		}
 	}
+	result = isc_dir_chroot(chroot_dir);
+	if (result != ISC_R_SUCCESS) {
+		fprintf(stderr, "isc_dir_chroot: %s\n",
+			isc_result_totext(result));
+		exit(1);
+	}
+	result = isc_dir_chdir("/");
+	if (result != ISC_R_SUCCESS) {
+		fprintf(stderr, "isc_dir_chdir: %s\n",
+			isc_result_totext(result));
+		exit(1);
+	}
 
 	if (argv[isc_commandline_index] != NULL)
 		conffile = argv[isc_commandline_index];
