--- bind-9.2.2rc1/isc-config.sh.in.orig	2001-04-17 02:29:55 +0400
+++ bind-9.2.2rc1/isc-config.sh.in	2003-02-08 22:39:16 +0300
@@ -108,7 +108,7 @@
 	echo $exec_prefix
 fi
 if test x"$echo_cflags" = x"true"; then
-	includes="-I${exec_prefix}/include"
+	[ "$exec_prefix" = /usr ] || includes="-I${exec_prefix}/include"
 	if test x"$libisc" = x"true"; then
 		includes="$includes @ALWAYS_DEFINES@ @STD_CINCLUDES@ @STD_CDEFINES@ @CCOPT@"
 	fi
