diff -uprk.orig bind-9.2.4rc5.orig/isc-config.sh.in bind-9.2.4rc5/isc-config.sh.in
--- bind-9.2.4rc5.orig/isc-config.sh.in	2004-03-09 09:09:08 +0300
+++ bind-9.2.4rc5/isc-config.sh.in	2004-06-30 17:11:54 +0400
@@ -108,7 +108,7 @@ if test x"$echo_exec_prefix" = x"true" ;
 	echo $exec_prefix
 fi
 if test x"$echo_cflags" = x"true"; then
-	includes="-I${exec_prefix}/include"
+	[ "$exec_prefix" = /usr ] || includes="-I${exec_prefix}/include"
 	if test x"$libisc" = x"true"; then
 		includes="$includes @ALWAYS_DEFINES@ @STD_CINCLUDES@ @STD_CDEFINES@ @CCOPT@"
 	fi
